Cython/Includes/cpython/context.pxd

```diff
@@ -0,0 +1,63 @@
+from .object cimport PyObject
+
+cdef extern from "Python.h":
+
+    ###########################################################################
+    # Context Variables Objects
+    ###########################################################################
+
+    bint PyContext_CheckExact(object o)
+    bint PyContextVar_CheckExact(object o)
+    bint PyContextToken_CheckExact(object o)
+
+    object PyContext_New()
+    # Return value: New reference.
+    # Create a new empty context object. Returns NULL if an error has occurred.
+
+    object PyContext_Copy(object ctx)
+    # Return value: New reference.
+    # Create a shallow copy of the passed ctx context object. Returns NULL if
+    # an error has occurred.
+
+    object PyContext_CopyCurrent()
+    # Return value: New reference.
+    # Create a shallow copy of the current thread context. Returns NULL if an
+    # error has occurred.
+
+    int PyContext_Enter(object ctx)
+    # Set ctx as the current context for the current thread. Returns 0 on
+    # success, and -1 on error.
+
+    int PyContext_Exit(object ctx)
+    # Deactivate the ctx context and restore the previous context as the
+    # current context for the current thread. Returns 0 on success, and -1 on
+    # error.
+
+    int PyContext_ClearFreeList()
+    # Clear the context variable free list. Return the total number of freed
+    # items. This function always succeeds.
+
+    object PyContextVar_New(const char *name, object default)
+    # Return value: New reference.
+    # Create a new ContextVar object. The name parameter is used for
+    # introspection and debug purposes. The default parameter may optionally
+    # specify the default value for the context variable. If an error has
+    # occurred, this function returns NULL.
+
+    int PyContextVar_Get(object var, object default_value, PyObject **value)
```

Comment:
This is a bit of a tricky function, because it allows `NULL` arguments for object parameters, and passing those will actually be a common use case. Thus, `object` is not the right type here, but `PyObject*` (which would allow `NULL`) is annoying because it requires users to cast their object values first.

What about defining macro aliases (as a C code docstring to this `cdef extern` block), e.g.
```C
#define __Pyx_PyContextVar_GetNoDefault(object var, PyObject **value) PyContextVar_Get(var, NULL, value)
```
and then declare it to Cython as
```cython
int PyContextVar_GetNoDefault "__Pyx_PyContextVar_GetNoDefault"(object var, PyObject **value) except -1 
```

Reply:
Hmm, okay.